lib/adapters/show-document-adapter.ts

```diff
@@ -0,0 +1,59 @@
+import { shell } from "electron"
+import { LanguageClientConnection, ShowDocumentParams, ShowDocumentResult } from "../languageclient"
+import { TextEditor } from "atom"
+import Convert from "../convert"
+
+/** Public: Adapts the window/showDocument command to Atom's text editors or external programs. */
+export default class ShowDocumentAdapter {
+  /** {@inheritDoc attach} */
+  public attach(arg: Parameters<typeof attach>[0]): ReturnType<typeof attach> {
```

Comment:
Should these functions be static?

Reply:
They can, but in my opinion, static methods are fundamentally wrong, because they do not use anything from the class, and the class is just being misused as a namespace!

To be honest, I have just kept this class to be consistent with other adapters, otherwise, I would have removed it.

Reply:
> because they do not use anything from the class, and the class is just being misused as a namespace!

I think that is kind of the point. These function don't use anything from the class. The class is just a namespace. I don't see what the problem is with that.

I feel like something like:
```js
import ShowDocumentAdapter from '...'
...
ShowDocumentAdapter.attach(...)
```

is better than:

```js
import {attach} from '...'
...
attach(...) // where does this non-namespaced attach come from?
```

Namespacing is very useful.

Reply:
I don't say that namespacing is not useful, but JavaScript and TypeScript both support namespaces, and there is no need to use a class as a namespace.
```ts
import * as ns from "...."
ns.attach()
```
or
```js
const ns = require("...")
ns.attach()
```

Reply:
I found an easier way to maintain consistency.
https://github.com/atom-community/atom-languageclient/pull/135/commits/96926433a19296e94e7df49cddc72119edced873

Reply:
I still think these should be consistent with other adapters.

The biggest reason is so we don't have to figure out if it is supposed to be `import ... from ...` or `import * as ... from ...` Which one works for each adapter? It would be better if they were all the same.

Reply:
The way I have `export default` here, make `import ... from ...` working like other adapters.

Reply:
> there is no need to use a class as a namespace.

I could argue that there is no need to *not* use a class as a namespace. Is there a good reason to maintain both ways? I feel like it is just extra maintenance without any benefit.

Reply:
This is just a simple `export default { export1, export2 } `. There is no maintenance cost involved with re-exporting.

Reply:
What you just said is extra maintenance by definition. Not a lot of extra maintenance, granted, but still extra maintenance for no benefit. 

I guess we can merge it this way and maybe make it consistent in another PR down the road.